ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -922,6 +936,186 @@ prop_diffusion_target_active_public defaultBearerInfo diffScript =
 
+prop_diffusion_target_active_local :: AbsBearerInfo
+                                   -> DiffusionScript
+                                   -> Property
+prop_diffusion_target_active_local defaultBearerInfo diffScript =
+    let sim :: forall s . IOSim s Void
+        sim = diffusionSimulation (toBearerInfo defaultBearerInfo)
+                                  diffScript
+                                  tracersExtraWithTimeName
+                                  tracerDiffusionSimWithTimeName
+
+        events :: [Events DiffusionTestTrace]
+        events = fmap ( Signal.eventsFromList
+                      . fmap (\(WithName _ (WithTime t b)) -> (t, b))
+                      )
+               . Trace.toList
+               . splitWithNameTrace
+               . Trace.fromList ()
+               . fmap snd
+               . Trace.toList
+               . fmap (\(WithTime t (WithName name b)) -> (t, WithName name (WithTime t b)))
+               . withTimeNameTraceEvents
+                  @DiffusionTestTrace
+                  @NtNAddr
+               . Trace.fromList (MainReturn (Time 0) () [])
+               . fmap (\(t, tid, tl, te) -> SimEvent t tid tl te)
+               . take 125000
+               . traceEvents
+               $ runSimTrace sim
+
+     in conjoin
+      $ (\ev ->
+        let evsList = eventsToList ev
+            lastTime = fst
+                     . last
+                     $ evsList
+         in classifySimulatedTime lastTime
+          $ classifyNumberOfEvents (length evsList)
+          $ verify_target_active_local ev
+        )
+      <$> events
+  where
+    verify_target_active_local :: Events DiffusionTestTrace -> Property
+    verify_target_active_local events =
+        let govLocalRootPeersSig :: Signal (Set NtNAddr)
+            govLocalRootPeersSig =
+              selectDiffusionPeerSelectionState
+                (LocalRootPeers.keysSet . Governor.localRootPeers) events
+
+            govActivePeersSig :: Signal (Set NtNAddr)
+            govActivePeersSig =
+              selectDiffusionPeerSelectionState Governor.activePeers events
+
+            localInActive :: Signal Bool
+            localInActive =
+              (\localPeers active ->
+                Set.size
+                (localPeers `Set.intersection` active)
+                  > 0
+              ) <$> govLocalRootPeersSig
+                <*> govActivePeersSig
+
+            meaning :: Bool -> String
+            meaning False = "No LocalPeers in Active Set"
+            meaning True  = "LocalPeers in Active Set"
+
+            valuesList :: [String]
+            valuesList = map (meaning . snd)
+                       . Signal.eventsToList
+                       . Signal.toChangeEvents
+                       $ localInActive
+
+         in checkCoverage
+          $ coverTable "active local peers"
+                       [("LocalPeers in Active Set", 1)]
+          $ tabulate "active local peers" valuesList
+          $ True
```

Comment:
Not quite because `localInActive` can have `False` values, meaning at that point no peer from the local set was active and we want to know the percentage of time there was local peers in active.

Reply:
Oh, you're absolutely right :smile: